apps/root/src/pages/earn/frame/index.tsx

```diff
@@ -4,23 +4,41 @@ import EarnFAQ from '../components/faq';
 import { useAppDispatch } from '@state/hooks';
 import { changeRoute } from '@state/tabs/actions';
 import { EARN_ROUTE } from '@constants/routes';
-import AllStrategiesTable from '../components/all-strategies-table';
+import AllStrategiesTable from '../components/strategies-table';
 import EarnWizard from '../components/wizard';
+import useEarnService from '@hooks/earn/useEarnService';
+import useFilteredStrategies from '@hooks/earn/useFilteredStrategies';
+import { StrategyColumnKeys } from '../components/strategies-table/components/columns';
 
 const EarnFrame = () => {
   const dispatch = useAppDispatch();
+  const earnService = useEarnService();
+  const { strategies, hasFetchedAllStrategies } = useFilteredStrategies();
 
   React.useEffect(() => {
     dispatch(changeRoute(EARN_ROUTE.key));
   }, []);
 
+  React.useEffect(() => {
+    const fetchStrategies = async () => {
+      await earnService.fetchAllStrategies();
+    };
+    if (!hasFetchedAllStrategies) {
```

Comment:
Related to the comment below, same here, if we've got this on this frame, hasFetchedAllStrategies will trigger a re-render in all child components, maybe the check only needs to go into the fetchAllStrategies method of the earnService

Reply:
About this 3 comments, I based a lot of this PR into making this component as flexible as possible! This 3 props would change from the user strategies table.
About the flag itself, I think it's necessary at this point as an isLoading state. We could move it to the service though, but this useEffect check would be still here